megamek/src/megamek/server/totalWarfare/TWDamageManagerModular.java

```diff
@@ -3046,15 +3049,18 @@ public int applyEntityArmorDamage(Entity entity, HitData hit, int damage, boolea
                 entity.damageThisPhase += damage;
 
                 damage = 0;
-                if (!entity.isHardenedArmorDamaged(hit)) {
-                    report = new Report(6085);
+                // Use trooper-specific messages for Battle Armor
+                if (entity instanceof BattleArmor) {
```

Comment:
Have separate BA and Infantry handling _here_ is probably okay, as this method is called for all unit types.